pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -173,52 +209,47 @@ func (w *Watcher) Start() error {
 		return err
 	}
 
-	watcher, err := fsnotify.NewWatcher()
+	fsWatcher, err := fsnotify.NewWatcher()
 	if err != nil {
-		return fmt.Errorf("failed to start plugin watcher, err: %v", err)
+		return fmt.Errorf("failed to start plugin fsWatcher, err: %v", err)
 	}
+	w.fsWatcher = fsWatcher
 
-	if err := watcher.Add(w.path); err != nil {
-		watcher.Close()
-		return fmt.Errorf("failed to start plugin watcher, err: %v", err)
+	if err := fsWatcher.Add(w.path); err != nil {
```

Comment:
Move `fsWatcher.Add(...)` calls inside `traversePluginsDir`.  This is because you only going to add a path during successful traversal anyway.

Reply:
we handle fsWatcher.Add() failure differently during Watcher.Start() and after it is started, so we can not put them together, thanks

Reply:
`w.path` as root is also `Walk`ed, so it will be added twice, no? I feel this `Watcher.Add` should only appear once or twice in the whole patch.